src/panels/lovelace/special-rows/hui-call-service-row.ts

```diff
@@ -1,25 +1,58 @@
-import { html } from "@polymer/polymer/lib/utils/html-tag.js";
-import { PolymerElement } from "@polymer/polymer/polymer-element.js";
+import { html, LitElement } from "@polymer/lit-element";
 import "@polymer/paper-button/paper-button.js";
 
 import "../../../components/ha-icon.js";
+
 import callService from "../common/call-service.js";
+import { EntityRow, CallServiceConfig } from "../entity-rows/types.js";
+import { HomeAssistant } from "../../../types.js";
+import { TemplateResult } from "lit-html";
+
+class HuiCallServiceRow extends LitElement implements EntityRow {
+  public hass?: HomeAssistant;
+  private _config?: CallServiceConfig;
+
+  static get properties() {
+    return {
+      hass: {},
+      _config: {},
+    };
+  }
+
+  public setConfig(config: CallServiceConfig): void {
+    if (
+      !config ||
+      !config.icon ||
+      !config.name ||
+      !config.action_name ||
+      !config.service ||
+      !config.service_data
+    ) {
+      throw new Error("Error in card configuration.");
+    }
+    this._config = config;
+  }
+
+  protected render(): TemplateResult {
+    if (!this._config) {
+      return html``;
+    }
 
-class HuiCallServiceRow extends PolymerElement {
-  static get template() {
     return html`
-      ${this.styleTemplate}
-      <ha-icon icon="[[_config.icon]]"></ha-icon>
+      ${this.renderStyle()}
+      <ha-icon icon="${this._config.icon}"></ha-icon>
       <div class="flex">
         <div>
-          [[_config.name]]
+          ${this._config.name}
         </div>
-        <paper-button on-click="_callService">[[_config.action_name]]</paper-button>
+        <paper-button
+          @click="${() => callService(this._config, this.hass)}"
```

Comment:
Instead of doing this, can you make it a private function so we don't have to create a new arrow function on each render.

Reply:
Is that just more overhead to create the arrow functions?

Reply:
`render` is called many times, each time we define a function. That function, just being generated, won't be unique to the previous one, causing lit to update the click listener. (`oldVal !== newVal`). It also means that you create a function per instance of the element.

Now with a class method, the definition of the method just lives on the prototype chain. There is only 1 in memory even if you have 1000 instances. No updates are triggered when setting (`oldVal === newVal`). And as added bonus, if someone would want to inherit from this class, they could override the method. Overall a lot better 👍